packages/core/crates/core-hopr/src/lib.rs

```diff
@@ -46,16 +56,30 @@ use {
 
 const MAXIMUM_NETWORK_UPDATE_EVENT_QUEUE_SIZE: usize = 2000;
 
-#[cfg(feature = "wasm")]
-#[wasm_bindgen]
 #[derive(Clone)]
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+pub struct ChannelEventEmitter {
+    tx: UnboundedSender<ChannelEntry>,
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+impl ChannelEventEmitter {
+    pub async fn send_event(&self, channel: &ChannelEntry) {
```

Comment:
It'd be better to pass channel entry by value, since it is cloned either way.

Reply:
It's passed by reference because it is a wasm-bound method, that will otherwise nullify the instance in TS (if passed by value).
So better keep it as reference and clone it in Rust.